examples/bridge-app/esp32/main/main.cpp

```diff
@@ -0,0 +1,452 @@
+{
+    uint16_t endpointIndex = emberAfGetDynamicIndexFromEndpoint(endpoint);
+
+    EmberAfStatus ret = EMBER_ZCL_STATUS_FAILURE;
+
+    // ChipLogProgress(DeviceLayer, "emberAfExternalAttributeWriteCallback: ep=%d", endpoint);
+
+    if (endpointIndex < DYNAMIC_ENDPOINT_COUNT)
+    {
+        Device * dev = gDevices[endpointIndex];
+
+        if ((dev->IsReachable()) && (clusterId == ZCL_ON_OFF_CLUSTER_ID))
+        {
+            ret = HandleWriteOnOffAttribute(dev, attributeMetadata->attributeId, buffer);
+        }
+        else if (clusterId == ZCL_DESCRIPTOR_CLUSTER_ID)
+        {
+            ret = HandleWriteDescriptorAttribute(endpointIndex, attributeMetadata, buffer, attributeMetadata->size, index);
+        }
+    }
+
+    return ret;
+}
+
+void HandleDeviceStatusChanged(Device * dev, Device::Changed_t itemChangedMask)
+{
+    if (itemChangedMask & Device::kChanged_Reachable)
+    {
+        uint8_t reachable = dev->IsReachable() ? 1 : 0;
+        emberAfReportingAttributeChangeCallback(dev->GetEndpointId(), ZCL_BRIDGED_DEVICE_BASIC_CLUSTER_ID,
+                                                ZCL_REACHABLE_ATTRIBUTE_ID, CLUSTER_MASK_SERVER, 0, ZCL_BOOLEAN_ATTRIBUTE_TYPE,
+                                                &reachable);
+    }
+
+    if (itemChangedMask & Device::kChanged_State)
+    {
+        uint8_t isOn = dev->IsOn() ? 1 : 0;
+        emberAfReportingAttributeChangeCallback(dev->GetEndpointId(), ZCL_ON_OFF_CLUSTER_ID, ZCL_ON_OFF_ATTRIBUTE_ID,
+                                                CLUSTER_MASK_SERVER, 0, ZCL_BOOLEAN_ATTRIBUTE_TYPE, &isOn);
+    }
+
+    if (itemChangedMask & Device::kChanged_Name)
+    {
+        uint8_t zclName[kUserLabelSize];
+        ToZclCharString(zclName, dev->GetName(), kUserLabelSize - 1);
+        emberAfReportingAttributeChangeCallback(dev->GetEndpointId(), ZCL_BRIDGED_DEVICE_BASIC_CLUSTER_ID,
+                                                ZCL_USER_LABEL_ATTRIBUTE_ID, CLUSTER_MASK_SERVER, 0, ZCL_CHAR_STRING_ATTRIBUTE_TYPE,
+                                                zclName);
+    }
+    if (itemChangedMask & Device::kChanged_Location)
+    {
+        uint8_t buffer[kFixedLabelAttributeArraySize];
+        EmberAfAttributeMetadata am = { .attributeId  = ZCL_LABEL_LIST_ATTRIBUTE_ID,
+                                        .size         = kFixedLabelAttributeArraySize,
+                                        .defaultValue = nullptr };
+
+        EncodeFixedLabel("room", dev->GetLocation(), buffer, sizeof(buffer), &am);
+
+        emberAfReportingAttributeChangeCallback(dev->GetEndpointId(), ZCL_FIXED_LABEL_CLUSTER_ID, ZCL_LABEL_LIST_ATTRIBUTE_ID,
+                                                CLUSTER_MASK_SERVER, 0, ZCL_ARRAY_ATTRIBUTE_TYPE, buffer);
+    }
+}
+
+extern "C" void app_main()
+{
+    int err = 0;
+    // Initialize the ESP NVS layer.
+    err = nvs_flash_init();
+    if (err != CHIP_NO_ERROR)
```

Comment:
nvs_flash_init returns esp_err_t. So you should check the return value with `ESP_OK`